### PR TITLE
[WIP] Introduced a permission to read system configuration.

### PR DESCRIPTION
**Pull request by kohsuke**
... similar to that of Item.EXTENDED_READ.

This is useful for Jenkins whose configuration is managed externally,
such as configuration as code.

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

### Proposed changelog entries

* Introduced a new permission that can read system configuration but not overwrite it

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

@jenkinsci/code-reviewers

### CODE REVIEW

**Comment by kohsuke** [comment]
@oleg-nenashev nod on the general direction would be appreciated before I get too far

File: core/src/main/java/jenkins/management/PluginsLink.java [hunk 1]
```diff
@@ -26,6 +26,8 @@
 
 import hudson.Extension;
 import hudson.model.ManagementLink;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/management/PluginsLink.java]: Huh?

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 2]
```diff
@@ -5064,6 +5064,11 @@ private static void computeVersion(ServletContext context) {
 
     public static final PermissionGroup PERMISSIONS = Permission.HUDSON_PERMISSIONS;
     public static final Permission ADMINISTER = Permission.HUDSON_ADMINISTER;
+    /**
+     * Permission to read configuration.
+     * @see Item#EXTENDED_READ
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/Jenkins.java]: Huh?

File: core/src/main/resources/jenkins/management/Messages.properties [hunk 3]
```diff
@@ -22,7 +22,7 @@
 # THE SOFTWARE.
 #
 
-ConfigureLink.DisplayName=Configure System
+ConfigureLink.DisplayName=System Configuration
```
Comment (daniel-beck) [inline core/src/main/resources/jenkins/management/Messages.properties]: Plenty of documentation will become outdated with this change.

**Comment by oleg-nenashev** [comment]
`Jenkins.EXTENDED_READ` should be probably considered as Dangerous permission. Reason: owner of this permission can probably see everything on the instance, including the information required to decrypt credentials (proof?). Needs deeper review IMO once the scope of the permission is fully defined

CC @daniel-beck @Wadeck .

File: core/src/main/resources/jenkins/management/Messages.properties (hunk 3 above)
Comment (kohsuke) [reply core/src/main/resources/jenkins/management/Messages.properties]: OK, I can keep the text intact for users with the `ADMINISTER` permission

File: core/src/main/java/jenkins/model/Jenkins.java (hunk 2 above)
Comment (kohsuke) [reply core/src/main/java/jenkins/model/Jenkins.java]: I'm afraid I don't understand what you are implying.

**Comment by jglick** [comment]
Is this really necessary? We already have more permissions than we can coherently explain. I would rather keep on checking `ADMINISTER` for these purposes. If you want to make certain configuration be treated as read-only, because it is generated, that should be done with a different API, rather than (ab)using permissions.

**Comment by jglick** [comment]
For example, a flag that can be set either on a particular `Item`, or on `Jenkins` overall, which says that it is generated and thus we should forbid any write operation which would normally require `CONFIGURE` (or `DELETE`, for items). That would allow us to get rid of some of the hacks we have in `branch-api` + `workflow-multibranch` as well, and be useful for `job-dsl` and probably other cases.

**Comment by oleg-nenashev** [comment]
No plan to review it anytime soon. The PR is also in progress && @jglick's comments have not been addressed yet

**Comment by daniel-beck** [comment]
It doesn't look like this will be successful in its current form, and there's been no progress basically since this was initially proposed more than a year ago. Therefore closing this PR.

**Pull request closed without merging**